### PR TITLE
Minimax heuristic

### PR DESCRIPTION
**Pull request by Michalski-Artur**
- [x] Implement minimax player using heuristic based on number of lines can go through the cell

2 bugs fixed:
- [x] Checking 4 in a row 
- [x] Incrementing number_of_wins when backpropagating in mcts

### CODE REVIEW

File: player/minimax/minimax_player.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            return max_evaluation
+        else:
+            min_evaluation = math.inf, None
+            for move in game_state.available_moves:
+                evaluation = self.minimax(game_state, current_depth+1, True, move, is_first_player)
+                if evaluation[0] < min_evaluation[0]:
+                    min_evaluation = evaluation[0], move
+            return min_evaluation
+
+    def evaluate_state(self, game_state, is_first_player):
```
Comment (holdenkold) [inline player/minimax/minimax_player.py]: add return type to a function

File: player/minimax/minimax_player.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    # TODO:
+    #  If its hardcoded, it cannot be used for other board sizes, maybe it should be generated?
+    @property
+    def heuristic_fields_values(self):
+        return self._heuristic_fields_values
+
+    def get_and_make_next_move_for_player(self, game_state: IGameState) -> int:
+        game_state_copy = game_state.clone()
+        is_first_player = game_state.is_first_player_move
+        evaluation, move = self.minimax(game_state_copy, 0, True, None, is_first_player)
```
Comment (holdenkold) [inline player/minimax/minimax_player.py]: could we add named arguments here? like current_depth=0 etc.?

File: player/minimax/minimax_player.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        game_state.make_move(move)
+        return move
+
+    def minimax(self, game_state, current_depth, max_turn, move, is_first_player):
```
Comment (holdenkold) [inline player/minimax/minimax_player.py]: add return type to a function

**Review by holdenkold — APPROVED** [review]

File: player/minimax/minimax_player.py (hunk 2 above)
Comment (Michalski-Artur) [reply player/minimax/minimax_player.py]: image

File: player/minimax/minimax_player.py (hunk 1 above)
Comment (Michalski-Artur) [reply player/minimax/minimax_player.py]: Done!

File: player/minimax/minimax_player.py (hunk 2 above)
Comment (Michalski-Artur) [reply player/minimax/minimax_player.py]: Added!

File: player/minimax/minimax_player.py (hunk 3 above)
Comment (Michalski-Artur) [reply player/minimax/minimax_player.py]: Done!

**Pull request merged**